src/middlewares/__tests__/createMetadataMiddleware.ts

```diff
@@ -110,45 +112,220 @@ describe('createMetadataMiddleware', () => {
 
       expect(JSON.parse(document.head.querySelector('meta')!.content))
         .toMatchInlineSnapshot(`
+        Object {
+          "widgets": Array [
+            Object {
+              "params": Array [],
+              "type": "ais.searchBox",
+              "widgetType": "ais.searchBox",
+            },
+            Object {
+              "params": Array [],
+              "type": "ais.searchBox",
+              "widgetType": "ais.searchBox",
+            },
+            Object {
+              "params": Array [
+                "escapeHTML",
+              ],
+              "type": "ais.hits",
+              "widgetType": "ais.hits",
+            },
+            Object {
+              "params": Array [],
+              "type": "ais.index",
+              "widgetType": "ais.index",
+            },
+            Object {
+              "params": Array [],
+              "type": "ais.pagination",
+              "widgetType": "ais.pagination",
+            },
+            Object {
+              "params": Array [
+                "searchParameters",
+              ],
+              "type": "ais.configure",
+              "widgetType": "ais.configure",
+            },
+          ],
+        }
+      `);
+    });
+
+    describe('fills it with user agent after start', () => {
+      it('for the v4 client', async () => {
+        const fakeSearchClient = createSearchClient();
+        const searchClient = algoliasearch('', '');
+        // @ts-expect-error overriding the search method for testing
+        searchClient.search = fakeSearchClient.search;
+
+        // not using createMetadataMiddleware() here,
+        // since metadata is built into instantsearch
+        const search = instantsearch({
+          searchClient,
+          indexName: 'test',
+        });
+
+        search.start();
+
+        await wait(100);
+
+        expect(document.head).toMatchInlineSnapshot(`
+          <head>
+            <meta
+              content="{\\"widgets\\":[],\\"ua\\":\\"Algolia for JavaScript (4.8.6); Node.js (12.14.1); instantsearch.js (4.17.0); JS Helper (3.4.4)\\"}"
```

Comment:
Can we check these with regex with only actual number on the major?
Every time we update the version of any of them, this test will fail.

Reply:
solved in afc78a6